core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -246,15 +246,18 @@ public void reloadConfig() {
   public void start() {
```

Comment:
`start` now creates a thread if one is not present and also stops the listeners. This is to prevent the general case of failed listeners and a sync thread that is not null when a network listener only reports that it is back online again, not offline. That's how the android one works. This doesn't cause any issues AFAIK.